pangeo_forge_recipes/transforms.py

```diff
@@ -460,19 +474,39 @@ def expand(self, reference: beam.PCollection) -> beam.PCollection:
         )
 
 
+class SampleSingleton(beam.PTransform):
+    """Receive an input PCollection of any size, sample a single value from it,
+    and emit a singleton PCollection containing the single sampled value.
+    """
+
+    def expand(self, pcoll: beam.PCollection) -> beam.PCollection:
+        return (
+            pcoll
+            | beam.combiners.Sample.FixedSizeGlobally(1)
+            | beam.FlatMap(lambda x: x)  # https://stackoverflow.com/a/47146582
+        )
+
+
 @dataclass
 class StoreToZarr(beam.PTransform, ZarrWriterMixin):
     """Store a PCollection of Xarray datasets to Zarr.
 
-    :param combine_dims: The dimensions to combine
-    :param target_chunks: Dictionary mapping dimension names to chunks sizes.
-        If a dimension is a not named, the chunks will be inferred from the data.
+     :param combine_dims: The dimensions to combine
+     :param target_chunks: Dictionary mapping dimension names to chunks sizes.
+         If a dimension is a not named, the chunks will be inferred from the data.
+     :param consolidate_dimension_coordinates: Whether to rewrite coordinate variables as a
+     single chunk. We recommend consolidating coordinate variables to avoid
+     many small read requests to get the coordinates in xarray. Defaults to ``True``.
+    :param consolidate_metadata: Whether to consolidate metadata in the resulting
+         Zarr dataset. Defaults to ``True``.
     """
 
     # TODO: make it so we don't have to explicitly specify combine_dims
     # Could be inferred from the pattern instead
     combine_dims: List[Dimension]
     target_chunks: Dict[str, int] = field(default_factory=dict)
+    consolidate_coords: bool = True
+    consolidate_metadata: bool = True
```

Comment:
Interesting @sbquinlan. Is there an example that could illustrate the advantage of exposing these stages? Just looking at this I would think this looks good since in almost all my cases I would want to have those two steps happening by default, but I might be missing something here.

Reply:
I think putting it that way makes a lot of sense @sbquinlan.